# Example 8.2
The distance from origin to point $P(4.5, 5.5)$ is given by $a=\sqrt{50.50}=7.1063$. Carry out an error analysis of this value if the coordinates of P have an error of $\pm 0.2$. How precise can one calculate the distance if it is known that the point is located on the line $x+y=10$?

In [1]:
from math import *
from numpy import *

In [2]:
# Given quantities
x = 4.5; y = 5.5       # point coordinates
a = sqrt(x**2 + y**2)  # distance origin->point
E_c = 0.2              # error limits for coordinates

We use the error propagation law for the error analysis.
The distance of the point from the origin is given by $$a = \sqrt{x^2+y^2}.$$

In [3]:
def dist(x,y):
    return sqrt(x**2 + y**2)

The error can be calculated as follows
$$\begin{align*}
E_a & = E_c \cdot \left ( \left | \frac{\partial a}{\partial x} \right | + \left | \frac{\partial a}{\partial y} \right | \right ) \\
 & = \frac{E_c ( x + y )}{\sqrt{x^2 + y^2}}.
\end{align*}$$

In [4]:
E_a = E_c*abs((x+y)/a); "a = %.1f  E_a = %.1f" % (dist(x,y),E_a)

'a = 7.1  E_a = 0.3'

The distance of the point P from origin can be specified therefor as $a=7.1 \pm 0.3$.

With the assumption $x+y=10$, the calculation of the point's distance becomes $$a = \sqrt{x^2+(10-x)^2}.$$


In [5]:
def dist(x):
    return sqrt(x**2+(10-x)**2)

The error can be calculated as follows by means of the error propagation law
$$\begin{align*}
E_a & = E_c \cdot \frac{\mathrm{d}a}{\mathrm{d}x} \\
 & = \frac{E_c(2x-10)}{\sqrt{x^2+(10-x)^2}}.
\end{align*}$$

In [6]:
E_a = E_c*abs((2*x-10)/dist(x)); "a = %.2f  E_a = %.2f" % (dist(x),E_a)

'a = 7.11  E_a = 0.03'

The distance of the point P from origin can be specified therefor as $a=7.11 \pm 0.03$.

We carry out an experimental error analysis for the assumption $x+y=10$ as a second solution variant.

In [7]:
# Range for x values
E_a = max(abs(dist(x-0.2)-dist(x)),abs(dist(x+0.2)-dist(x)))
"a = %.2f  E_a = %.2f" % (dist(x),E_a)

'a = 7.11  E_a = 0.03'

Therefore, the distance of the point P from origin can be specified as $a=7.11 \pm 0.03$.